### ISSUE TITLE
Preferences->General crashes

### ISSUE THREAD

**Issue by olwalter** [standardnotes/forum#1484]
Whenever I open Preferences -> General, the web app crashes. It is unresponsive until I reload the web page. In the web console, I get error below. Identical behavior in the standalone app. Any help would be appreciated. This is an extended account.

Uncaught (in promise) TypeError: e.name is undefined

app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2 GET
(anonymous) @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
runRequest @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
runHttp @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
tokenRefreshableRequest @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
getSetting @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
getSensitiveSetting @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
getSensitiveSetting @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
isMfaActivated @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
isMfaActivated @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
fetchStatus @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
Te @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
fetchStatus @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
t.TwoFactorAuthWrapper @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
R @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
O @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
E @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
j @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
O @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
E @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
j @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
O @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
E @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
j @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
O @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
E @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
O @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
E @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
O @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
E @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
O @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
(anonymous) @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
C @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
Promise.then (async)
_ @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
I.setState @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
o.__c.o.__ @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
a @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
(anonymous) @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
t.runReaction_ @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
Bt @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
Ct @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
(anonymous) @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
V @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
Ct @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
Et @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
mt @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
Fe @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
Te @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
selectPane @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
onClick @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
onClick @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
D @ app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
Show 2 more frames
app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'toLowerCase')
    at app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
    at Array.sort (<anonymous>)
    at app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
    at app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
    at app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2
    at et (app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2)
    at e.t.track (app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2)
    at _ (app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2)
    at I.c [as constructor] (app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2)
    at I.R [as render] (app-777d9d6ffd49653bb2afb784d5709605d5bddd510fc0a7f03f58d7b48b5f4216.js:2)

DevTools failed to load source map: Could not load content for https://app.standardnotes.org/assets/javascripts/app.js.map;: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE

**Comment by moughxyz** [comment]
This is because you have a component with an undefined name. We could add a safety check to not try to access name if it doesn't exist, but I'm more curious where a component with no name could have come from. Do you dabble with installing custom extensions?

Can you download a decrypted backup file of your data and filter for items with content_type SN|Component, then find the component that is missing a name, and share that blob with me?

**Comment by olwalter** [comment]
I downloaded an unencrypted backup as a zip and checked the text files in Items/SN_Components. They all have names in it. 

*** SN_Component-104b8e1a.txt ***
  "name": "Task Editor",
    "name": "Task Editor",
    "permission_name": "editor:task-editor",
      "name": "stream-context-item"
*** SN_Component-14705c0b.txt ***
  "name": "Secure Spreadsheets",
    "name": "Secure Spreadsheets",
    "permission_name": "editor:sheets",
[...]

I also checked the large JSON file in my backup folder and jq'ed the SN|Components. They all look similar, except the first, which lacks the line
  "duplicate_of": null

I played around with the extensions the other day, but not with custom extensions, just the ones that are available in my account.
